README.md

```diff
@@ -22,27 +22,25 @@ If you need a logger, you can use the interface like this:
 ```php
 <?php
 
+declare(strict_types=1);
+
 use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
 
-class Foo
+final readonly class Foo
 {
-    private $logger;
+    public function __construct(
+        private LoggerInterface $logger = new NullLogger()
+    ){}
 
-    public function __construct(LoggerInterface $logger = null)
+    public function doSomething(): void
     {
-        $this->logger = $logger;
-    }
+        $this->logger->info('Doing work');
```

Comment:
If you want to do modern PHP, IMO it's best to leave $logger nullable and call it conditionally with:

```suggestion
        $this->logger?->info('Doing work');
```

That avoids useless method calls and a useless NullLogger instance.

Reply:
Depends. 

That means:
- having your code riddled with "?->"
- depending on a static code analysis tool (like PHPStan or Psalm) to prevent nullpointer exceptions

I think this is more a question of deprecating "NullLogger" or not (which is out of scope for this PR).

Personally, I like to keep it simple and stick to OOP with DI, since it is language agnostic.

Reply:
I don't see why this package should deprecate NullLogger. Both NullLogger and the nullsafe operator are valid ways to solve the case of the optional logger in the constructor

> Personally, I like to keep it simple and stick to OOP with DI, since it is language agnostic.

being language agnostic does not make sense when writing a code snippet for PHP.

And if you mean avoiding features that don't directly translate to other OOP languages, your argument is void IMO:
- you are using constructor promoted properties, which have a direct equivalent in very few other languages
- the nullsafe operator exists in most modern languages

Reply:
Is it a recommended way to handle with no logs - pass a `null` , not `NullLogger`?